src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -403,105 +403,124 @@ exprt string_constraint_generatort::add_axioms_for_correct_number_format(
   axioms.push_back(
     implies_exprt(correct, str.axiom_for_is_shorter_than(max_size)));
 
-  // forall 1 <= qvar < |str| . correct => is_digit_with_radix(str[qvar], radix)
-  symbol_exprt qvar=fresh_univ_index("number_format", index_type);
-  exprt is_digit=is_digit_with_radix(str[qvar], radix);
-  string_constraintt all_digits(
-    qvar, from_integer(1, index_type), str.length(), correct, is_digit);
-  axioms.push_back(all_digits);
+//  // forall 1 <= qvar < |str| . correct => is_digit_with_radix(str[qvar], radix)
+//  symbol_exprt qvar=fresh_univ_index("number_format", index_type);
+//  exprt is_digit=is_digit_with_radix(str[qvar], radix);
+//  string_constraintt all_digits(
+//    qvar, from_integer(1, index_type), str.length(), correct, is_digit);
+//  axioms.push_back(all_digits);
+
+  for(std::size_t index=1; index<max_size; ++index)
```

Comment:
The constraint for index 0 is slightly different, because the first character can be `+` or `-` or a digit. This is handled earlier in the function.